runtime/vm/jvminit.c

```diff
@@ -4222,6 +4222,13 @@ processVMArgsFromFirstToLast(J9JavaVM * vm)
 			vm->extendedRuntimeFlags2 |= J9_EXTENDED_RUNTIME2_CPU_LOAD_COMPATIBILITY;
 		}
 	}
+	{
+		IDATA enableExtendedHCR =  FIND_AND_CONSUME_VMARG(EXACT_MATCH, VMOPT_XXENABLEEXTENDEDHCR, NULL);
+		IDATA disableExtendedHCR =  FIND_AND_CONSUME_VMARG(EXACT_MATCH, VMOPT_XXDISABLEEXTENDEDHCR, NULL);
+		if (disableExtendedHCR >= enableExtendedHCR) {
+			vm->extendedRuntimeFlags2 |= J9_EXTENDED_RUNTIME2_DISABLE_EXTENDED_HCR;
+		}
```

Comment:
You should probably add the else case here so the option works properly when. both positive and negative options are specified (which might well be the case if you take Pete's suggestion for IBM java 8).

Reply:
I think this would be better if it followed the existing pattern (look at flight recorder below). Set the bit at the top of this function to default to no extended and use > and < tests rather than >= .

Reply:
Actually I'm thinking about this wrong (thinking that each option goes through this code, which is not the case). The consume code gives the largest index at which the option is found, so the code is fine as it is.

Reply:
With the current code, you will need to set the bit at the top of this function to have it default to disabled when no options are specified.